### ISSUE TITLE
Blitz Conflict

### ISSUE THREAD

**Issue by daniellelecomte** [a-digital/cookie-consent-banner#20]
Hey there! This seems to have a conflict with the Blitz caching plugin. When Blitz is active, the cookie banner never displays using the auto-inject method. The code never even makes it into the DOM. 

Have you seen this or have a possible solution?

Thanks!

**Comment by mark-adigital** [comment]
Hi - I can't say I've used Blitz, but having just taken a look at it, the problem seems likely to be that the version of the page which is in the Blitz cache doesn't include the banner for some reason. This could be due to the banner having already been dismissed by yourself or another user when the page was requested and added to the cache, or the way in which Blitz requests the page might be tripping one of several conditionals I've added to try and ensure the banner is only rendered when it should be.

It's possible tweaking these might mean the markup and JS/CSS resources would be included in pages from Blitz cache, but could throw something else out and introduces another performance issue - that those assets and markup would be included on each page load even if it wasn't displayed, whether or not the banner had already been dismissed. Presently if the banner has already been dismissed, none of that occurs.

I'm open to discussing this further and amending the Cookie Consent Banner code if necessary, but given this is a free plugin with no charge for support/updates and Blitz is a commercial one, I hope you don't find it unreasonable for me to suggest you start by asking this question in the Blitz repository instead - Ben Croker is much better placed than me to suggest why the way in which Blitz generates its cache may be preventing the banner from rendering and then either resolve the issue from within Blitz, or at least provide additional info for me to patch the Cookie Banner.
